Source/Common/ValuesJsonConverter.cs

```diff
@@ -383,11 +354,19 @@ private static bool TryProcessTokenAsType(JsonReader reader, Type targetType, ou
                     result = localResult;
                 }
             }
-            else if (tokenType is JsonToken.Integer or JsonToken.Float)
+            else if (tokenType == JsonTokenType.Number)
+            {
+                if (targetType == typeof(short) || targetType == typeof(int) || targetType == typeof(long) || targetType == typeof(float) || targetType == typeof(double) || targetType == typeof(decimal))
```

Comment:
Should we be checking for all number types here? Including unsighned types?

Reply:
Maybe? We don't have any `Values` types on our models that have unsigned values but people who extend Schema.NET might.